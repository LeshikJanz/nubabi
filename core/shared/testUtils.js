import configureMockStore from 'redux-mock-store';
import { combineEpics, createEpicMiddleware } from 'redux-observable';

export const setupEpic = (epic) => {
  const epicMiddleware = createEpicMiddleware(epic);
  const mockStore = configureMockStore([epicMiddleware]);

  return {
    middleware: epicMiddleware,
    mockStore: mockStore,
    afterEach: () => {
      epicMiddleware.replaceEpic(epic);
    }
  }
};
