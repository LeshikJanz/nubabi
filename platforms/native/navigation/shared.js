import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) =>
  pathOr('Nubabi', ['params', 'babyName'], state);

export function getBabyTitle(navigation) {
  if (!navigation) {
    throw new Error('an screen hasnt been updated, check stack');
  }

  return babyNameTitle(navigation);
}

export const getTabHeaders = navigate => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
