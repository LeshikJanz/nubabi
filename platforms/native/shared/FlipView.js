// @flow
import React, { Component } from 'react';
import { View, Animated } from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

type Props = {
  renderFront: Function,
  renderBack: Function,
  velocity: number,
  tension: number,
  friction: number,
};

export class FlipCard extends Component {
  props: Props;

  state = {
    animatedValue: new Animated.Value(0),
    isFlipped: true,
  };

  componentDidUpdate(prevProps: Props, prevState: *) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard = () => {
    this.setState({ isFlipped: !this.state.isFlipped });
  };

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0, // Returns to the start
      velocity: this.props.velocity, // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction, // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront(this._flipToggleCard);
    }
    return this.props.renderBack(this._flipToggleCard);
  }

  render() {
    const perspective = 2000;
    const rotateY = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <View style={styles.animatedContainer}>
        <Animated.View
          style={[
            styles.animatedContainer,
            { transform: [{ rotateY }, { perspective }] },
          ]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </View>
    );
  }
}

export default FlipCard;
