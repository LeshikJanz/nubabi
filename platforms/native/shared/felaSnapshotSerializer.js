// WIP
// prettier-ignore
const felaSnapshotSerializer = {
  test(val) {
    return val && val.felaRules && val.$$typeof === Symbol.for('react.test.json');
  },

  print(val, print) {
    const styles = val.felaRules;
    val.felaRules = false;

    return `style = Object ${JSON.stringify(styles, null, 4)}\n\n${print(val)}`;
  },
};

module.exports = felaSnapshotSerializer;
