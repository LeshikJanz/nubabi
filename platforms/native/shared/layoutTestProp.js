export default {
  fontScale: 2,
  height: 1334,
  scale: 2,
  width: 750,
  viewportWidth: 750,
  viewportHeight: 1334,
};
