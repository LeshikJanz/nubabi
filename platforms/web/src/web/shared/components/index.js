import About from "./About";
import AsyncComponent from "./AsyncComponent";
import AuthenticatedRoute from "./AuthenticatedRoute";
import Header from "./Header";
import Home from "./Home";
import Loader from "./Loader";
import Test from "./Test";

export default {
  About,
  AsyncComponent,
  AuthenticatedRoute,
  Header,
  Home,
  Loader,
  Test
};
