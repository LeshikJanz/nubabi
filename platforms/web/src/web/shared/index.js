import * as actions from "./actions";
export { default as containers } from "./containers";
export { default as reducer } from "./reducer";
export { default as epics } from "./epics";
export { actions };
