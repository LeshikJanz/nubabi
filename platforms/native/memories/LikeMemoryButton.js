// @flow
import type { ToggleMemoryLikeInput } from 'core/types';
import React from 'react';
import { LayoutAnimation, TouchableOpacity } from 'react-native';
import { gql } from 'react-apollo';
import { compose, propOr } from 'ramda';
import { hoistStatics, withHandlers } from 'recompose';
import { Box, Icon, Text } from '../components';
import theme from 'core/themes/defaultTheme';

type Props = {
  id: string,
  isLikedByViewer: boolean,
  onToggleLike: (input: ToggleMemoryLikeInput) => Promise<ApolloQueryResult<*>>,
  onToggle: (isLiked: boolean) => void,
  likes?: { count: number },
};

export const LikeMemoryButton = ({
  onToggle,
  isLikedByViewer: isLiked,
  likes,
  withCount = false,
}: Props) => {
  const iconName = isLiked ? 'md-heart' : 'md-heart-outline';
  const iconColor = isLiked ? theme.colors.primary : theme.colors.secondary;
  const likeCount = propOr(0, ['count'], likes);

  return (
    <Box flexDirection="row">
      <Box as={TouchableOpacity} onPress={onToggle}>
        <Icon
          name={iconName}
          size={16}
          color={iconColor}
          style={{ marginRight: 5 }}
        />
      </Box>
      {likeCount > 0 && withCount && <Text color="secondary">{likeCount}</Text>}
    </Box>
  );
};

LikeMemoryButton.fragments = {
  item: gql`
    fragment LikeMemoryButton on Memory {
      id
      isLikedByViewer
      likes {
        count
        edges {
          actor {
            firstName
          }
        }
      }
    }
  `,
};

export default hoistStatics(
  compose(
    withHandlers({
      onToggle: ({ id, onToggleLike, isLikedByViewer, likes }) => () =>
        onToggleLike(id, !isLikedByViewer, likes.count),
    }),
  ),
)(LikeMemoryButton);
