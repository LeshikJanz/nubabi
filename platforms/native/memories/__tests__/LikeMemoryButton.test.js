import 'react-native';
import React from 'react';
import LikeMemoryButton from '../LikeMemoryButton';
import { expectRender } from '../../shared/testUtils';

describe('LikeMemoryButton', () => {
  it('renders correctly when liked by viewer', () => {
    expectRender(
      <LikeMemoryButton
        isLikedByViewer
        likes={{count: 5}}
      />,
    );
  });

  it('renders correctly when not liked by viewer', () => {
    expectRender(
      <LikeMemoryButton
        isLikedByViewer={false}
        likes={{count: 5}}
      />
    );
  });

  it('renders correctly when requested to add count', () => {
    expectRender(
      <LikeMemoryButton
        isLikedByViewer
        likes={{count: 5}}
        withCount
      />
    );
  });

  it('does not render like count even when requested if count is 0', () => {
    expectRender(
      <LikeMemoryButton
        isLikedByViewer
        likes={{count: 0}}
        withCount
      />
    );
  });
});
