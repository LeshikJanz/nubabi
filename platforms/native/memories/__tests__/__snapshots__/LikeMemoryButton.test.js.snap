// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeMemoryButton does not render like count even when requested if count is 0 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "flexDirection": "row",
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#748294",
    "fontFamily": "SF Pro Text"
}

<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#EA3154",
            "fontSize": 16,
          },
          Object {
            "marginRight": 5,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`LikeMemoryButton renders correctly when liked by viewer 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "flexDirection": "row"
}

<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#EA3154",
            "fontSize": 16,
          },
          Object {
            "marginRight": 5,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`LikeMemoryButton renders correctly when not liked by viewer 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "flexDirection": "row"
}

<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#748294",
            "fontSize": 16,
          },
          Object {
            "marginRight": 5,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`LikeMemoryButton renders correctly when requested to add count 1`] = `
style = Object {
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "flexDirection": "row",
    "fontSize": 13,
    "lineHeight": 16,
    "color": "#748294",
    "fontFamily": "SF Pro Text"
}

<View
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#EA3154",
            "fontSize": 16,
          },
          Object {
            "marginRight": 5,
          },
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopWidth": 0,
        "color": "#748294",
        "fontFamily": "SF Pro Text",
        "fontSize": 13,
        "lineHeight": 16,
      }
    }
  >
    5
  </Text>
</View>
`;
